lib/Glimmer.pm

```diff
@@ -0,0 +1,494 @@
+# use strict;
+# use Data::Dumper;
+
+# my $out = Glimmer::call_genes_with_glimmer(file_with_path, { verbose => 1 });
+
+# print Dumper($out);
+
+package Glimmer;
+use strict;
+use File::Slurp;
+use File::Temp;
+use Data::Dumper;
+
+sub call_genes_with_glimmer
+{
+    my($contigs_file, $opts) = @_;
+
+    my $genetic_code     =   $opts->{genetic_code} || 11;   #...Default to the "standard" code
+
+    my $train            = qq();   #...Flag associated with '-train' switch
+    my $training_tbl     = qq();
+    my $training_contigs = qq();
+    
+    my $min_training_len = $opts->{min_training_len} || 2000;   #...Shortest contig used for self-training
+    my $verbose = $opts->{verbose};
+
+    my $glimmeropts = qq(-o50 -g110 -t30 -l);   #...NOTE: Make these the default for a switch
+
+    my $genetic_code_switch = qq(-z $genetic_code);
+
+    my $id_prefix = $opts->{id_prefix} || "prot.";
+
+    if (! $training_contigs) { $training_contigs = $contigs_file }
+
+    my $tmpdir = File::Temp->newdir();
+    print "tmpdir=$tmpdir\n";
+    
+    my $tmp_prefix = "$tmpdir/glimmer";
+
+    my $tmp_contig       = "$tmp_prefix.contig";
+    my $tmp_coords       = "$tmp_prefix.coords";
+    my $tmp_train        = "$tmp_prefix.train";
+    my $tmp_model        = "$tmp_prefix.icm";
+    
+    my ($contig_id, $seqP);
+    
+    if (not $train) {
+	#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+	print STDERR "\nFinding training ORFs using GLIMMER default procedure\n"
+	    if $verbose;
+	#-----------------------------------------------------------------------    
+	print STDERR "Training file is $training_contigs\n" if $verbose;
+	open(CONTIGS, "<$training_contigs")
+	    || die "could not read-open training contigs file: $training_contigs";
```

Comment:
You should always use the three-argument form of open -- see https://perldoc.perl.org/5.32.0/open.html for examples.

Reply:
e.g.
``` perl
open (my $contigs_fh, "<", $training_contigs)
   or die "Could not open training contigs file $training_contigs: $!";
```

`$!` contains the reason why the file cannot be opened.